src/vm/readytoruninfo.cpp

```diff
@@ -348,7 +348,9 @@ BOOL ReadyToRunInfo::IsReadyToRunEnabled()
     WRAPPER_NO_CONTRACT;
 
     static ConfigDWORD configReadyToRun;
-    return configReadyToRun.val(CLRConfig::EXTERNAL_ReadyToRun);
+    return configReadyToRun.val(CLRConfig::EXTERNAL_ReadyToRun) && 
+            !(CORProfilerDisableAllNGenImages()) && 
```

Comment:
I think it would be better to have these checks in `ReadyToRunInfo::Initialize` where all sorts of the conditions are checked.